### ISSUE TITLE
Form field personalization

### ISSUE THREAD

**Issue by lucas-gymex** [sulu/SuluFormBundle#234]
| Q | A
| --- | ---

Hello,

I am currently finalizing our website, and am looking to customize my drop-down list.

I succeeded for the other fields, as we can see on the following code:
```
{% extends 'SuluFormBundle:themes:dynamic.html.twig' %}

{%- block sulu_form_row_class_name -%}
    {%- set width = attr.width|default('full') -%}
    {%- if width == 'full' -%}
        {%- set width = 'col-md-12' -%}
    {%- elseif width == 'one-third' -%}
        {%- set width = 'col-md-4' -%}
    {%- endif -%}

    {%- set className = '' ~ width ~ ' form-group' -%}
    {{- className -}}
{%- endblock sulu_form_row_class_name -%}

{%- block form_row -%}
    {{ block('sulu_form_row_start') }}
    {{- form_widget(form) -}}
    {{- form_label(form) -}}
    {{- form_errors(form) -}}
    {{ block('sulu_form_row_end') }}
{%- endblock -%}

{%- block form_widget_simple -%}
    {%- set attr = attr|merge({class: 'form-control'}) -%}
    {{ parent() }}
{%- endblock form_widget_simple -%}

{%- block textarea_widget -%}
    {%- set attr = attr|merge({class: 'form-control'}) -%}
    {{ parent() }}
{%- endblock textarea_widget -%}

{%- block submit_widget -%}
    {%- set attr = attr|merge({class: 'btn btn-primary'}) -%}
    <div class="text-center">
        {{ parent() }}
    </div>
{%- endblock submit_widget -%}
```

However, I don't know what the **widget** for the drop-down list is.
If someone could guide me or tell me what the name of this widget is so that I can move forward, it would be great!

thank you

**Comment by alexander-schranz** [comment]
@lucas-gymex the themes are outputted the default symfony way you find the basic symfony form theme here: 

https://github.com/symfony/symfony/blob/master/src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig

best thing to get all available `block_prefixes` is doing a `dump` in b:

```twig
{%- block form_row -%}
    {{ dump(form.vars.block_prefixes) }}

    {{ block('sulu_form_row_start') }}
    {{- form_widget(form) -}}
    {{- form_label(form) -}}
    {{- form_errors(form) -}}
    {{ block('sulu_form_row_end') }}
{%- endblock -%}
```

every block_prefix can be used to overwrite something e.g.`{% block _dynamic_form_email_widget %}`

So basically following block names are possible for every type:

 - BLOCKPREFIX_row
 - BLOCKPREFIX_widget
 - BLOCKPREFIX_label

The dropdown choice type (choice in symfony is basically select, multi select, radio, checkboxes) to overwrite the dropdown only you should go with `choice_widget_collapsed` but have a look at `form_div_layout.html.twig`

https://github.com/symfony/symfony/blob/0bec08f0d8005a9a20de13ca27e1f316a7b13a53/src/Symfony/Bridge/Twig/Resources/views/Form/form_div_layout.html.twig#L58-L78

**Comment by lucas-gymex** [comment]
thank you for your answer, you helped me a lot
